### ISSUE TITLE
Huggingface BERT prediction issues

### ISSUE THREAD

**Issue by lucala** [amaiya/ktrain#384]
```
def predict(self, val_data=None):
        """
        ```
        Makes predictions on validation set
        ```
        """
        if val_data is not None:
            val = val_data
        else:
            val = self.val_data
        if val is None: raise Exception('val_data must be supplied to get_learner or predict')
        if hasattr(val, 'reset'): val.reset()
        classification, multilabel = U.is_classifier(self.model)
        preds = self.model.predict(self._prepare(val, train=False))
        if hasattr(preds, 'logits'): # dep_fix: breaking change in transformers==4.0.0 - also needed for Longformer
        #if type(preds).__name__ == 'TFSequenceClassifierOutput': # dep_fix: undocumented breaking change in transformers==4.0.0
            # REFERENCE: https://discuss.huggingface.co/t/new-model-output-types/195
            preds = preds.logits

        # dep_fix: transformers in TF 2.2.0 returns a tuple insead of NumPy array for some reason
        if isinstance(preds, tuple) and len(preds) == 1: preds = preds[0] 

        if classification:
            if multilabel:
                return activations.sigmoid(tf.convert_to_tensor(preds)).numpy()
            else:
                return activations.softmax(tf.convert_to_tensor(preds)).numpy()
        else:
            return preds
```
Hi, I trained a multilingual BERT Model from Huggingface via Ktrain for binary classification. When I want to make predictions the following happens:
`self._prepare(val, train=False)` returns a working tfdataset (contains entries as expected) - but `self.model.predict(self._prepare(val, train=False))` then returns `TFSequenceClassifierOutput(loss=None, logits=None, hidden_states=None, attentions=None)` where the problem occurs with `preds = preds.logits`. Since `preds` is now None it throws an exception when trying to convert to a tensor at the end.

I have transformers version 4.3.3 and Ktrain version 0.27.2. Any idea what could be the problem?

**Comment by amaiya** [comment]
Can you please tell me precisely which model you're using, so I can try to reproduce the problem?     I wasn't able to reproduce this problem using a `bert-base-multilingual-cased` model.  

Also, what does your training set look like (i.e., in what format are your labels)?

**Comment by lucala** [comment]
I seem to have found the problem. I'm not sure why this solved the issue but I accidently still had a multi-gpu flag set in my code while running only on one GPU. Scoping the classifier inside `tf.distribute.MirroredStrategy()` seems to have caused the issues even though the data looked alright.

For anyone wondering, if prediction was successful the object should look something like this
```
TFSequenceClassifierOutput(loss=None, logits=array([[-0.08186093, -0.08343899],
       [-0.07120603, -0.12185383],
       [-0.08030262, -0.14472717],
       [-0.08776484, -0.09777661],
       [-0.08230339, -0.12103553],
       [-0.09109291, -0.08862668],
       [-0.09338085, -0.05343446],
       [-0.10880657, -0.07758594],
       [-0.1216483 , -0.10442323],
       [-0.09323623, -0.07971355],
       [-0.08031276, -0.05662249],
       [-0.09421368, -0.08208254],
       [-0.05550822, -0.11889505],
       [-0.08128081, -0.13111375],
       [-0.07634296, -0.12988581],
       [-0.10191501, -0.06580244],
       [-0.06480942, -0.07017672],
       [-0.11083872, -0.060646  ],
       [-0.08702252, -0.10050216],
       [-0.07380851, -0.05861536],
       [-0.11353912, -0.06525099],
       [-0.05151699, -0.09708275],
       [-0.04874814, -0.09655503],
       [-0.06435809, -0.12561524],
       [-0.07859507, -0.05844422],
       [-0.08236887, -0.06153866],
       [-0.07638695, -0.06774613],
       [-0.08463045, -0.07762618],
       [-0.11477844, -0.08476482],
       [-0.05005918, -0.14127994],
       [-0.07722785, -0.11858568],
       [-0.05399485, -0.06812211]], dtype=float32), hidden_states=None, attentions=None)

```
